torch/testing/__init__.py

```diff
@@ -212,7 +212,8 @@ def assert_allclose(actual, expected, rtol=None, atol=None, equal_nan=True, msg=
     if not isinstance(expected, torch.Tensor):
         expected = torch.tensor(expected, dtype=actual.dtype)
     if expected.shape != actual.shape:
-        expected = expected.expand_as(actual)
+        raise AssertionError("expected tensor shape {0} doesn't match with actual tensor "
```

Comment:
this is in fact causing lots of failure in fbcode. @mruberry should I proceed to provide a warning instead?

Reply:
Making it a warning sounds OK for now. When we start to document torch.testing (soon) we'll probably want to update those callsites in FBCode.

Reply:
fixed all test errors internally instead (there weren't that many afterall :-) )